### PR TITLE
Respect new strict parsing of `email.utils.getaddresses`

### PR DESCRIPTION
**Pull request by petschki**
preferred solution for #4020 ... successor of #4021

### CODE REVIEW

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

File: Products/CMFPlone/PloneTool.py [hunk 1]
```diff
@@ -142,6 +142,10 @@ def validateSingleNormalizedEmailAddress(self, address):
             # Address contains two newlines (possible spammer relay attack)
             return False
 
+        # address can be empty if getaddresses has parsing errors (returns [("", "")])
```
Comment (mauritsvanrees) [inline Products/CMFPlone/PloneTool.py:145]: Can we do `address = address.strip()` first, for good measure?  Actually I would do that a few lines above, after the `isinstance` check.

When we are being called by our own `validateSingleEmailAddress` there is probably nothing to strip away, but someone else may call this method.

**Review by mauritsvanrees — COMMENTED** [review]
Thanks for working on this.  I have a few suggestions.
`PloneTool.py` has these methods:

* `validateEmailAddresses`: validate a string with possibly multiple email addresses.  For each address this calls `validateSingleNormalizedEmailAddress`.
* `validateSingleEmailAddress`: validate a string with at most ONE email address.  For this address this calls `validateSingleNormalizedEmailAddress`.
* `validateSingleNormalizedEmailAddress` only needs to handle one address, obviously.

- [x] In all three methods: after checking if it is a string, can we do a `.strip()`?
- [x] In the two methods handling only a single address, can we do `address.splitlines()` and exit when this returns more than one line?  In both cases this can replace the use of `EMAIL_CUTOFF_RE`.

BTW, I agree with your removing this line from the valid inputs:

```
 user@example.org\n user2@example.org",  # omitting comma is ok
```

File: Products/CMFPlone/PloneTool.py (hunk 1 above)
Comment (petschki) [reply Products/CMFPlone/PloneTool.py:145]: yes good point ... I'll change that.

**Comment by petschki** [comment]
> BTW, I agree with your removing this line from the valid inputs:
> 
> ```
>  user@example.org\n user2@example.org",  # omitting comma is ok
> ```

I first moved this line to the invalid inputs but then it fails on current py3.10 jenkins with no strict parsing.

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

**Comment by petschki** [comment]
Currently its hard to get a green jenkins ...

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

**Comment by rohnsha0** [comment]
@jenkins-plone-org please run jobs

**Comment by rohnsha0** [comment]
Is Node4 experiencing flakiness again, or could the issues be related to code changes?

**Comment by petschki** [comment]
@rohnsha0 please see progress here #4019

**Comment by rohnsha0** [comment]
Ah, I understood... thanks for providing more info!

**Comment by petschki** [comment]
Currently the build is broken because of this strict parsing which got backported to 3.10 too (see https://jenkins.plone.org/job/plone-6.1-python-3.10/371/) ... anyone willing to approve here?

**Review by davisagli — APPROVED** [review]
LGTM, thanks for working on this

**Review by rohnsha0 — APPROVED** [review]
LGTM

**Pull request merged**